packages/fuels/src/cli/commands/build/buildSwayPrograms.ts

```diff
@@ -0,0 +1,59 @@
+import { spawn } from 'child_process';
+import { join } from 'path';
+
+import type { FuelsConfig } from '../../types';
+import { findPackageRoot } from '../../utils/findPackageRoot';
+import { debug, log, loggingConfig } from '../../utils/logger';
+
+type OnResultFn = () => void;
+type OnErrorFn = (reason?: number | Error) => void;
+
+export const onForcExit =
+  (onResultFn: OnResultFn, onErrorFn: OnErrorFn) => (code: number | null) => {
+    if (code) {
+      onErrorFn(code);
+      // process.exit()?
+    } else {
+      onResultFn();
+    }
+  };
+
+export const onForcError = (onError: OnErrorFn) => (error: Error) => {
+  onError(error);
+};
+
+export const buildSwayProgram = async (config: FuelsConfig, path: string) => {
+  debug('Building Sway program', path);
+
+  return new Promise<void>((resolve, reject) => {
+    const pkgRootDir = findPackageRoot();
+    const forcPath = join(pkgRootDir, 'node_modules', '.bin', 'fuels-forc');
+
+    const command = config.useBuiltinForc ? forcPath : 'forc';
```

Comment:
```suggestion
    const command = config.useBuiltinForc ? join(findPackageRoot(), 'node_modules', '.bin', 'fuels-forc') : 'forc';
```

I initially got confused because I thought `useBuiltinForc` wasn't used at all - almost wrote a comment about it - but then I saw it underneath `forcPath`. I'd _prefer_ it like in the suggestion above because there's no need to get `pkgRootDir` and `forcPath` if `config.useBuiltinForc` is `false`.

Reply:
I renamed the variable but kept the rest as it seems costless. https://github.com/FuelLabs/fuels-ts/pull/1291/commits/f737c93f5b0773e953fc1adc5908a8e12c5d99b9